CSharp/ArticleProjects/PersistOutputs/OutputFilesExample.cs

```diff
@@ -0,0 +1,131 @@
+            GettingStartedCommon.CreatePoolIfNotExistAsync(batchClient, pool).Wait();
+
+            CloudJob job = batchClient.JobOperations.CreateJob(jobId, new PoolInformation { PoolId = poolId });
+
+            // Commit the job to the Batch service
+            job.Commit();
+            Console.WriteLine($"Created job {jobId}");
+
+            CloudBlobContainer container = storageAccount.CreateCloudBlobClient().GetContainerReference(containerName);
+            container.CreateIfNotExists();
+
+            string containerSas = container.GetSharedAccessSignature(new SharedAccessBlobPolicy()
+            {
+                Permissions = SharedAccessBlobPermissions.Write,
+                SharedAccessExpiryTime = DateTimeOffset.UtcNow.AddDays(1)
+            });
+            string containerUrl = container.Uri.AbsoluteUri + containerSas;
+
+            // Create a series of simple tasks which dump the task environment to a file and then write random values to a text file
+            IEnumerable<CloudTask> tasks = Enumerable.Range(1, 20).Select(i =>
+                {
+                    var taskId = i.ToString().PadLeft(3, '0');
+                    return new CloudTask(taskId, $"cmd /v:ON /c \"echo off && set && (FOR /L %i IN (1,1,100000) DO (ECHO !RANDOM!)) > output.txt\"")
+                    {
+                        OutputFiles = new List<OutputFile>
+                        {
+                            new OutputFile(
+                                filePattern: @"..\std*.txt",
+                                destination: new OutputFileDestination(new OutputFileBlobContainerDestination(
+                                    containerUrl: containerUrl,
+                                    path: taskId)),
+                                uploadOptions: new OutputFileUploadOptions(
+                                    uploadCondition: OutputFileUploadCondition.TaskCompletion)),
+                            new OutputFile(
+                                filePattern: @"output.txt",
+                                destination: new OutputFileDestination(new OutputFileBlobContainerDestination(
+                                    containerUrl: containerUrl,
+                                    path: taskId + @"\output.txt")),
+                                uploadOptions: new OutputFileUploadOptions(
+                                    uploadCondition: OutputFileUploadCondition.TaskCompletion)),
+                        }
+                    };
+                }
+            );
+
+            batchClient.JobOperations.AddTask(jobId, tasks);
+            job.Refresh();
+            tasks = job.ListTasks();
+
+            Console.WriteLine($"All tasks added to job {job.Id}");
+            Console.WriteLine();
+
+            // Wait for the tasks to all complete
+            TaskStateMonitor monitor = batchClient.Utilities.CreateTaskStateMonitor();
+            monitor.WaitAll(tasks, TaskState.Completed, TimeSpan.FromMinutes(15));
```

Comment:
TSM has the advantage of being built-in but the disadvantage that you have to wait for _all_ tasks to complete before starting to download the output of the first task.  Customers often want to download results as they become available, so it would be good to show this on the OutputFiles path too.